psalm-baseline.xml

```diff
@@ -785,7 +785,7 @@
     </UndefinedInterfaceMethod>
   </file>
   <file src="lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php">
-    <InvalidArgument occurrences="1"/>
+    <InvalidArgument occurrences="2"/>
```

Comment:
Which new error did you encounter here? Is it possible to fix them?

Reply:
Its because of
https://github.com/doctrine/orm/blob/497ee166bdffe45dd85494fe1eb31d454fad363b/lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php#L268-L276

Since the `ClassMetadataInfo::setDiscriminatorColumn($columnDef)` expects for `$columnDef` an array with `enumType?: class-string<BackedEnum>|null` but the `(string)` conversion confuses psalm into treating it as `string`

```
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=269,col=25,title=InvalidArgument::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:269:25: InvalidArgument: Argument 1 of Doctrine\ORM\Mapping\ClassMetadata::setDiscriminatorColumn expects array{columnDefinition?: null|string, enumType?: class-string<BackedEnum>|null, fieldName?: string, length?: int, name: null|string, type?: string}|null, but array{columnDefinition: null|string, enumType: null|string, length: int, name: null|string, type: string} provided (see https://psalm.dev/004)
```

Reply:
All right.

Reply:
Btw, I do have some more ideas:

```php
$discrColumnAttribute = $classAttributes[Mapping\DiscriminatorColumn::class];
assert($discrColumnAttribute instanceof Mapping\DiscriminatorColumn);
```

This does resolve the issue, but it creates new one:

```php
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=271,col=88,title=RedundantCastGivenDocblockType::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:271:88: RedundantCastGivenDocblockType: Redundant cast to string given docblock-provided type (see https://psalm.dev/263)
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=272,col=88,title=RedundantCastGivenDocblockType::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:272:88: RedundantCastGivenDocblockType: Redundant cast to string given docblock-provided type (see https://psalm.dev/263)
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=273,col=90,title=RedundantCastGivenDocblockType::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:273:90: RedundantCastGivenDocblockType: Redundant cast to int given docblock-provided type (see https://psalm.dev/263)
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=274,col=100,title=RedundantCastGivenDocblockType::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:274:100: RedundantCastGivenDocblockType: Redundant cast to string given docblock-provided type (see https://psalm.dev/263)
::error file=lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php,line=275,col=92,title=RedundantCastGivenDocblockType::lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php:275:92: RedundantCastGivenDocblockType: Redundant cast to class-string<BackedEnum> given docblock-provided type (see https://psalm.dev/263)
```

However I think it should be safe to remove the casting and strong-type the `DiscriminatorColumn` fields. It will require some fixing in internal code, as sometimes I saw we assign `$length` as `string` instead of int, and we need to treat this as BC break, but since this targets 2.14 and the ultimate goal is to have strong typed everything by 3.0, I think we should do this. WDYT?

Reply:
Actually, let me fix this in 3.0 after merge, I will replace

https://github.com/doctrine/orm/blob/2bfcb6e5ab3d4f66090179a2f4e936511e64d175/lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php#L258-L265

with 

```php
assert($discrColumnAttribute instanceof Mapping\DiscriminatorColumn);

$metadata->setDiscriminatorColumn(
    [
        'name'             => $discrColumnAttribute->name ?? null,
        'type'             => $discrColumnAttribute->type ?? 'string',
        'length'           => $discrColumnAttribute->length ?? 255,
        'columnDefinition' => $discrColumnAttribute->columnDefinition ?? null,
        'enumType'         => $discrColumnAttribute->enumType ?? null,
    ]
);
```